blueoil/templates/lmnet/semantic_segmentation.tpl.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+from easydict import EasyDict
+import tensorflow as tf
+
+from lmnet.common import Tasks, get_color_map
+from lmnet.networks.segmentation.lm_segnet_v1 import LmSegnetV1Quantize
+from lmnet.datasets.camvid import CamvidCustom
+from lmnet.data_processor import Sequence
+from lmnet.pre_processor import (
+    Resize,
+    DivideBy255,
+    PerImageStandardization,
+)
+from lmnet.quantizations import (
+    binary_mean_scaling_quantizer,
+    linear_mid_tread_half_quantizer,
+)
+
+{% if data_augmentation %}from lmnet.data_augmentor import ({% for augmentor in data_augmentation %}
+    {{ augmentor[0] }},{% endfor %}
+){% endif %}
+
+IS_DEBUG = False
+
+NETWORK_CLASS = {{network_class}}
+DATASET_CLASS = type('DATASET_CLASS', ({{dataset_class}},), {{dataset_class_property}})
+
+IMAGE_SIZE = {{image_size}}
+BATCH_SIZE = {{batch_size}}
+DATA_FORMAT = "NHWC"
+TASK = Tasks.SEMANTIC_SEGMENTATION
+CLASSES = {{classes}}
+
+MAX_EPOCHS = {{max_epochs}}
+SAVE_STEPS = {{save_steps}}
+TEST_STEPS = {{test_steps}}
+SUMMARISE_STEPS = {{summarise_steps}}
+
+# distributed training
+IS_DISTRIBUTION = False
+
+# pretrain
+IS_PRETRAIN = False
+PRETRAIN_VARS = []
+PRETRAIN_DIR = ""
+PRETRAIN_FILE = ""
+
+# for debug
+# BATCH_SIZE = 2
+# SUMMARISE_STEPS = 1
+# IS_DEBUG = True
+
+PRE_PROCESSOR = Sequence([
+    Resize(size=IMAGE_SIZE),
+    {% if quantize_first_convolution %}DivideBy255(){% else %}PerImageStandardization(){% endif %}
+])
+POST_PROCESSOR = None
+
+NETWORK = EasyDict()
+NETWORK.OPTIMIZER_CLASS = {{optimizer_class}}
+NETWORK.OPTIMIZER_KWARGS = {{optimizer_kwargs}}
+NETWORK.LEARNING_RATE_FUNC = {{learning_rate_func}}
+NETWORK.LEARNING_RATE_KWARGS = {{learning_rate_kwargs}}
+
+NETWORK.IMAGE_SIZE = IMAGE_SIZE
+NETWORK.BATCH_SIZE = BATCH_SIZE
+NETWORK.DATA_FORMAT = DATA_FORMAT
+NETWORK.ACTIVATION_QUANTIZER = linear_mid_tread_half_quantizer
+NETWORK.ACTIVATION_QUANTIZER_KWARGS = {
+    'bit': 2,
+    'max_value': 2
+}
+NETWORK.WEIGHT_QUANTIZER = binary_mean_scaling_quantizer
+NETWORK.WEIGHT_QUANTIZER_KWARGS = {}
+NETWORK.LABEL_COLORS = get_color_map(len({{classes}}))
```

Comment:
The `NETWORK.LABEL_COLORS` parameter can be hidden if you change source code on segmentation `Base` class in `lmnet/networks/segmentation/base.py` like below

```
    def __init__(
            self,
            *args,
            label_colors=None,
            **kwargs
    ):
        super().__init__(
            *args,
            **kwargs,
        )
        if label_colors is None:
            self.label_colors = get_color_map(self.num_classes)
        else:
            self.label_colors = label_colors
```

Less parameter is good for maintenance I think.

Reply:
@ruimashita Thanks. Fixed.